test/erlfmt_markdown_SUITE.erl

```diff
@@ -0,0 +1,175 @@
+        "hello(mike, joe, robert)\n"
+        "```\n"
+        "ignore for now\n"
+        "```erlang unformatted\n"
+        "goodbye(mike, joe, robert)\n"
+        "```\n"
+        "The previous unformatted version, should match this formatted version\n"
+        "```erlang formatted\n"
+        "goodbye(mike, joe, robert)\n"
+        "```\n"
+        "This unformatted version, should match the first formatted version with the same key\n"
+        "```erlang unformatted key\n"
+        "hello(mike, joe, robert)\n"
+        "```\n",
+    check_markdown(S).
+
+check_markdown(Content) ->
+    Sections = string:split(Content, "```", all),
+    lists:foldl(fun check_section/2, {text, maps:new()}, Sections).
+
+check_section(_Text, {text, State}) ->
+    {code, State};
+check_section(Text, {code, State}) ->
+    [FirstLine, CurrentCode] = string:split(Text, "\n"),
+    Spec = string:split(FirstLine, " ", all),
+    case Spec of
+        [Erlang | [ToFormat | Keys]] ->
+            case string:prefix(Erlang, "erl") of
+                nomatch ->
+                    {text, State};
+                _ ->
+                    Key = string:join(Keys, " "),
+                    case maps:find(Key, State) of
+                        error ->
+                            case ToFormat of
+                                "formatted" ->
+                                    check_fmt(CurrentCode, CurrentCode),
+                                    {text, maps:put(Key, CurrentCode, State)};
+                                "unformatted" ->
+                                    {text, maps:put(Key, CurrentCode, State)}
+                            end;
+                        {ok, PreviousCode} ->
+                            case ToFormat of
+                                "formatted" ->
+                                    check_fmt(PreviousCode, CurrentCode);
+                                "unformatted" ->
+                                    check_fmt(CurrentCode, PreviousCode)
+                            end,
+                            {text, maps:remove(Key, State)}
+                    end
+            end;
+        _ ->
+            {text, State}
+    end.
+
+check_fmt(Unformatted, Expected) ->
+    NewlyFormatted = format(Unformatted, 80),
+    ?assertEqual(string:trim(NewlyFormatted), string:trim(Expected)).
+
+format(String, PageWidth) ->
+    try format_doc(String) of
+        Doc ->
+            Rendered = erlfmt_algebra:document_render(Doc, [{page_width, PageWidth}]),
+            unicode:characters_to_list(Rendered)
+    catch
+        _:_ ->
+            throw("Big Problem Parsing: " ++ String)
+    end.
+
+format_doc(String) ->
+    case erlfmt:read_forms_string("nofile", "f() ->\n" ++ String ++ ".") of
+        {ok, [{function, _, [{clause, _, _, empty, [Expr]}]}], []} ->
+            erlfmt_format:expr_to_algebra(Expr);
+        _ ->
+            {ok, [Form], []} = erlfmt:read_forms_string("nofile", String),
```

Comment:
I think this is something we'll need to do in general to support formatting files like `rebar.config`, but :+1: for now

Reply:
Yes, even this function failed to format

```erl
{plugins, [erlfmt]}.
```

and

```erl
{erlfmt, [
    write,
    {files, "{src,include,test}/*.{hrl,erl}"}
]}.
```

But that would be nice to add in a follow up commit.